### PR TITLE
gcp - added security policy

### PR DESCRIPTION
**Pull request by evhenyshaliov**
Use cases:

policies:
  - name: epam-gcp-armor_security_policy_adaptive_protection_enabled
    description: |
      Cloud Armor Security Policy have not Adaptive Protection enabled
    resource: gcp.security-policy
    filters:
      - not:
          - type: value
            key: adaptiveProtectionConfig.layer7DdosDefenseConfig.enable
            value: true
            op: eq
  - name: epam-gcp-armor_security_policy_have_deny_as_default_action
    description: |
      Cloud Armor Security Policy does not have deny as default action
    resource: gcp.security-policy
    filters:
      - type: value
        key: "rules[?priority==`2147483647`].action | {range: join(', ', @)}.range"
        value: "allow"
        op: eq

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/resources/securitypolicy.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+from c7n_gcp.provider import resources
+from c7n_gcp.query import QueryResourceManager, TypeInfo
+
+
+@resources.register('security-policy')
+class SecurityPolicy(QueryResourceManager):
+    """GC resource:
+        https://cloud.google.com/bigtable/docs/reference/admin/rest/v2/projects.instances"""
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/resources/securitypolicy.py:10]: doc link needs updating

File: tools/c7n_gcp/c7n_gcp/resources/securitypolicy.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+from c7n_gcp.provider import resources
+from c7n_gcp.query import QueryResourceManager, TypeInfo
+
+
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/resources/securitypolicy.py:6]: can you rename the module to armor.py to reflect underlying service, security policy is a bit generic and has potential overlap into iam, it seems like these are from the cloud armor service.

File: tools/c7n_gcp/c7n_gcp/resources/securitypolicy.py (hunk 1 above)
Comment (evhenyshaliov) [reply tools/c7n_gcp/c7n_gcp/resources/securitypolicy.py]: fixed. thanks

File: tools/c7n_gcp/c7n_gcp/resources/securitypolicy.py (hunk 2 above)
Comment (evhenyshaliov) [reply tools/c7n_gcp/c7n_gcp/resources/securitypolicy.py]: fixed

File: tools/c7n_gcp/c7n_gcp/resources/armor.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    """GC resource: https://cloud.google.com/compute/docs/reference/rest/v1/securityPolicies"""
+    class resource_type(TypeInfo):
+        service = 'compute'
+        version = 'v1'
+        component = 'securityPolicies'
+        scope_key = 'project'
+        name = id = 'id'
+        scope_template = '{}'
+        permissions = ('compute.securityPolicies.list',)
+        default_report_fields = ['displayName', 'expireTime']
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/resources/armor.py:19]: looks like app armor policies are supported by asset inventory

`compute.googleapis.com/SecurityPolicy`
per https://cloud.google.com/asset-inventory/docs/supported-asset-types

Comment (evhenyshaliov) [reply tools/c7n_gcp/c7n_gcp/resources/armor.py:19]: Fixed. Please check

**Comment by kapilt** [comment]
renamed to armor-policy, fixed metadata (reporting and id/name in particular were off) and added urn. id should refer to primary mechanism to refer to resource via api. there's a new test helper method check_report_fields for validating default report fields fwiw.

**Review by kapilt — APPROVED** [review]
lgtm

**Pull request merged**